libyang/data.py

```diff
@@ -224,6 +281,31 @@ def validate(
         if ret != 0:
             raise self.context.error("validation failed")
 
+    def diff(
+        self,
+        other: "DNode",
+        no_siblings: bool = False,
+        with_defaults: bool = False,
+    ) -> Iterator[DDiff]:
+        flags = diff_flags(no_siblings=no_siblings, with_defaults=with_defaults)
+        difflist = lib.lyd_diff(self.cdata, other.cdata, flags)
+
+        i = 0
+        while True:
+            dtype = difflist.type[i]
+            first = None
+            second = None
+            if dtype == lib.LYD_DIFF_END:
+                break
+            if difflist.first[i] != ffi.NULL:
+                first = self.new(self.context, difflist.first[i])
+            if difflist.second[i] != ffi.NULL:
+                second = self.new(self.context, difflist.second[i])
```

Comment:
I prefer to avoid comparing values to `ffi.NULL`. Could you rewrite this like:

```python3
first = self.new(self.context, difflist.first[i]) if difflist.first[i] else None
second = self.new(self.context, difflist.second[i]) if difflist.second[i] else None
```

Or a more verbose version of it if you prefer. If you use a shorter variable name for `difflist` (`dlist` maybe?) it may fit in single line statements.

Reply:
I was not sure if a <cdata 'void *' NULL> is falsy. It is.

Reply:
yeah looked even better and more comapact after the change! However black made it worse. I hate this tool so much.